yaptide/celery/utils/progress/fluka_monitor.py

```diff
@@ -153,3 +155,31 @@ def read_fluka_out_file(event: threading.Event,
     logging.info("Sending final update for task %d, simulated primaries %d", details.task_id,
                  progress_details.requested_primaries)
     send_task_update(details.simulation_id, details.task_id, details.update_key, up_dict)
+
+
+def read_fluka_out_file_offline(sim_dir: Path) -> tuple[int, int]:
+    """Reads fluka out file and returns number of simulated and requested primaries"""
+    simulated_primaries = 0
+    requested_primaries = 0
+    in_progress = False
+    out_file_path = sim_dir / "fl_sim001.out"
+    if not out_file_path.exists():
+        logging.error("Out file %s not found", out_file_path)
+        return simulated_primaries, requested_primaries
+    try:
+        with open(out_file_path, 'r') as f:
+            for line in f:
+                logging.debug("Parsing line: %s", line.rstrip())
+                if line.startswith(S_TERMINATED_FROM_OUTSIDE) or line.startswith(S_OK_OUT_COLLECTED):
+                    break
+                if in_progress:
+                    simulated_primaries, remaining_particles = parse_progress_remaining_line(line)
+                    if requested_primaries == 0:
+                        requested_primaries = simulated_primaries + remaining_particles
+                    in_progress = False
+                else:
+                    if line.startswith(S_OK_OUT_IN_PROGRESS):
+                        in_progress = True
```

Comment:
can you explain the logic here in some comment lines ?
I don't get why do you check for requested == 0 ? and then calculate `requested_primaries = simulated_primaries + remaining_particles`

Reply:
It is the way to calculate a requestred parameter value, because fluka out file doesn't contain a info about it inside of it. We have got only a simulated primaries and remaining, it is sufficient to calculate requested primaries. I want count it only once, so for this reason there is == 0.

Reply:
what you write isn't very obvious. It's the reason why you've written in such a way it's written. The reasons why the code is written such way should go in the comment. That's main purpose of comments